src/main/java/seedu/parser/Parser.java

```diff
@@ -0,0 +1,155 @@
+    };
+    public void process(String response) {
+        String command = response.split(" ")[0];
+        if (commandProcessor.containsKey(command)) {
+            String statement = removeFirstWord(response);
+            try {
+                commandProcessor.get(command).execute(statement, this);
+            } catch (IllegalArgumentException | IllegalStateException | SysLibException e) {
+                System.out.println(e.getMessage());
+            }
+        } else {
+            System.out.println("no commands found. Enter \"help\" for a list of commands." + SEPARATOR_LINEDIVIDER);
+        }
+
+    }
+    public static String removeFirstWord(String response) {
+        int index = response.indexOf(" ");
+        if (index == -1) {
+            return "";
+        }
+        return response.substring(index + 1);
+    }
+
+    public List<Resource> getResourceList() {
+        return resourceList;
+    }
+
+    public static String[] parseAddCommand(String statement) throws SysLibException {
+        String inputPattern = "(.+?) /tag (.)(.+)";
+
+        Pattern pattern = Pattern.compile(inputPattern);
+        Matcher matcher = pattern.matcher(statement);
+        boolean matchFound = matcher.find();
+
+        if (matchFound) {
+            if (matcher.group(2).equalsIgnoreCase("b")) {
+                return parseAddBook(statement);
+            } else {
+                throw new SysLibException("Please use the format " +
+                        "'add /id ID /t TITLE /a AUTHOR /tag TAG /i ISBN [/g GENRE]'." + SEPARATOR_LINEDIVIDER);
+            }
+        } else {
+            throw new SysLibException("Please use the format " +
+                    "'add /id ID /t TITLE /a AUTHOR /tag TAG /i ISBN [/g GENRE]'." + SEPARATOR_LINEDIVIDER);
+        }
+    }
+
+    public static String[] parseAddBook(String statement) throws SysLibException, IllegalStateException {
+        try {
+            String inputPattern = "/id (.+?) /t (.+?) /a (.+?) /tag (.+?) /i (.+)";
+            String genrePattern = "(.+) /g (.+)";
+
+            Pattern pattern = Pattern.compile(inputPattern);
+            Matcher matcher = pattern.matcher(statement);
+            boolean matchFound = matcher.find();
+
+            Pattern gPattern = Pattern.compile(genrePattern);
+            Matcher gMatcher = gPattern.matcher(matcher.group(5));
+            boolean gMatchFound = gMatcher.find();
+
+            String[] args = new String[6];
+
+            if (matchFound) {
+                args[0] = matcher.group(1).trim(); // id
+                args[1] = matcher.group(2).trim(); // title
+                args[2] = matcher.group(3).trim(); // author
+                args[3] = matcher.group(4).trim(); // tag
+                if (gMatchFound) {
+                    args[4] = gMatcher.group(1).trim(); // isbn
+                    args[5] = gMatcher.group(2).trim(); // genre
+                } else {
+                    args[4] = matcher.group(5).trim(); // isbn
```

Comment:
Magic literals (e.g. group(<5>)) - would be good if the numbers are placed as a named constant.